### PR TITLE
MISC FIXES in prep for Storybook 3 release

### PR DESCRIPTION
**Pull request by shilman**
Hi @lo1tuma I created a bunch of new "features" while prepping the Storybook3 release. I'm issuing this PR as an FYI so you can take a look. It's not intended to be merged (in fact, it shouldn't!). If you want me to split out some of these features as individual PRs, I can do that at some point.

- GH_TOKEN support
- skippable PR labels that don't show up in CHANGELOG (e.g. for fixing typos)
- smart collapsible PR groups
- accept multiple valid labels for a single PR

### CODE REVIEW

File: lib/createChangelog.js [hunk 1]
```diff
@@ -1,23 +1,45 @@
...
+    return formatted;
+}
 
-        pullRequests.forEach(function (pullRequest) {
-            changelog += `* ${pullRequest.title} (#${pullRequest.id})\n`;
-        });
+function createChangelog(newVersionNumber, formatOptions, mergedPullRequests) {
+    const { validLabels, skipLabels } = formatOptions;
+    const groupedPullRequests = R.groupBy(R.prop('label'), mergedPullRequests);
+    const date = moment().locale('en').format('YYYY-MMMM-DD');
```
Comment (lo1tuma) [inline lib/createChangelog.js]: Looks like you also changed the date format. We could make this a configurable option, what do you think?

File: lib/createChangelog.js [hunk 2]
```diff
@@ -1,23 +1,45 @@
...
 
-function createChangelog(newVersionNumber, validLabels, mergedPullRequests) {
-    const groupedPullRequests = R.groupBy(R.prop('label'), mergedPullRequests);
-    const date = moment().locale('en').format('MMMM D, YYYY');
-    const title = `## ${newVersionNumber} (${date})`;
+const COLLAPSE_COUNT = 10;
 
-    let changelog = `${title}\n\n`;
+function formatPR(pr) {
+    return `-   ${pr.title} [#${pr.id}](https://github.com/storybooks/storybook/pull/${pr.id})\n`;
```
Comment (lo1tuma) [inline lib/createChangelog.js]: 👍 for providing a link to the PR, but we should determine the github owner and repo automatically.

**Comment by lo1tuma** [comment]
Hi @shilman,

there are a lot of nice additions in this PR which I would be happy to merge. Also I think some of them should be behind a feature flag and not enabled by default:

* smart collapsible PR groups: there should be an option to enable this feature which should be disabled by default. Maybe it would even make sense to make the PR count configurable (currently hardcoded to `10`), but we can also do this later once it get requested.
* accept multiple valid labels for a single PR: I’m not sure what the right thing to do would be. AFAIU you just picked the first valid label and ignored the others. We could also add the PR to ALL valid labels, so it would appear multiple times in the changelog. I’ve also noticed that when there is no valid label you are setting the default `other` which might also not be the right thing to do. Sometimes I forget to set labels to a PR in such cases I’m really happy that `pr-log` gives me a very specific error message so I can add the labels afterwards. That’s being said, what ever we are doing here should be behind a feature flag and disabled by default, the explicit errors which we currently have should be the default behavior. 

The other changes are looking good so far 👍.

File: lib/getMergedPullRequests.js [hunk 3]
```diff
@@ -21,21 +21,14 @@ function getPullRequests(fromTag) {
 
             return mergeCommits
                 .map((commit) => commit.match(/^Merge pull request #(\d+) from (.*?) \((.*)\)$/u))
-                .filter((matches) => matches)
+                .filter((matches) => matches && parseInt(matches[1], 10) > 500)
```
Comment (lo1tuma) [inline lib/getMergedPullRequests.js]: I’m not sure if I understand what this change is doing.

Comment (shilman) [inline lib/getMergedPullRequests.js]: LOL commented in the commit log, but should have commented in the code. Anyway, it's temporary and very Storybook 3.0 specific. 😭

File: lib/createChangelog.js (hunk 2 above)
Comment (shilman) [inline lib/createChangelog.js]: Totally agree!

File: lib/createChangelog.js (hunk 1 above)
Comment (shilman) [inline lib/createChangelog.js]: 👍

**Pull request closed without merging**